test/parallel/test-webstorage.js

```diff
@@ -14,6 +14,19 @@ function nextLocalStorage() {
   return join(tmpdir.path, `${++cnt}.localstorage`);
 }
 
+test('constructor is illegal', async () => {
+  const cp = await spawnPromisified(process.execPath, [
+    '--experimental-webstorage',
+    '--localstorage-file', nextLocalStorage(),
+    '-e', 'new Storage',
+  ]);
+  assert.strictEqual(cp.code, 1);
+  assert.strictEqual(cp.signal, null);
+  assert.strictEqual(cp.stdout, '');
+  assert(cp.stderr.includes(`TypeError: Illegal constructor\n`));
+  assert(cp.stderr.includes(`code: 'ERR_ILLEGAL_CONSTRUCTOR'\n`));
```

Comment:
Can you use `assert.match` so in case of failure we get a more informative error message?
```suggestion
  assert.match(cp.stderr, /TypeError: Illegal constructor\n(.*\n)*\s*code: 'ERR_ILLEGAL_CONSTRUCTOR'\n/);
```